### PR TITLE
Feat/basic auth improvements

### PR DESCRIPTION
**Pull request by opaolini**
About
===

This PR adds `SigningSchemes` [0] which allows the `auth-es-proxy` to be configured which `SigningScheme` it should use. It also adds `BASIC_AUTH_SCHEME` as a signing option.

[0] - `Signing` doesn't seem 100% accurate, as in the case of Basic Auth we are not "signing" anything just adding an authentication header 🤔

### CODE REVIEW

File: basic_signer.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	}, nil
+
+}
+
+type BasicAuthSigner struct {
+	username string
+	password string
+}
+
+func (p *BasicAuthSigner) SignRequest(r *http.Request) error {
```
Comment (fabioberger) [inline basic_signer.go]: `p` doesn't seem like the appropriate abbreviation here.

File: integration_test.go [hunk 2]
```diff
@@ -161,3 +161,72 @@ func TestAllowedRegexpProxy(t *testing.T) {
...
 	es.Stop(ctx)
 }
+
+func TestBasicAuthRoundtrip(t *testing.T) {
+	tc := newTestClient()
+
+	echoServerAddr := "localhost:3010"
+	es := newEchoServer(echoServerAddr)
+
+	ctx := context.Background()
```
Comment (fabioberger) [inline integration_test.go]: Seems this `ctx` isn't used anywhere?

File: integration_test.go [hunk 3]
```diff
@@ -161,3 +161,72 @@ func TestAllowedRegexpProxy(t *testing.T) {
...
+	authProxyConfig := ProxyConfig{
+		RemoteAddress:              "http://" + echoServerAddr,
+		ShouldValidateRequests:     true,
+		AuthenticationScheme:       BasicAuthScheme,
+		AllowedBasicAuthUserString: fmt.Sprintf("%s:%s", username, password),
+	}
+	authProxyAddr := "localhost:3012"
+
+	proxy, err := NewProxy(&authProxyConfig)
+
```
Comment (fabioberger) [inline integration_test.go]: Get rid of unnecessary new lines.

File: integration_test.go [hunk 4]
```diff
@@ -161,3 +161,72 @@ func TestAllowedRegexpProxy(t *testing.T) {
...
+	require.NoError(t, err)
+
+	go func() {
+		t.Log(http.ListenAndServe(authProxyAddr, proxy))
```
Comment (fabioberger) [inline integration_test.go]: Will this go routine ever exit?

File: integration_test.go [hunk 5]
```diff
@@ -161,3 +161,72 @@ func TestAllowedRegexpProxy(t *testing.T) {
...
+		ShouldSignOutgoing: true,
+		SigningScheme:      BasicAuthScheme,
+		BasicAuthUser:      username,
+		BasicAuthPassword:  password,
+	}
+	signingProxyAddr := "localhost:3013"
+
+	signingProxy, err := NewProxy(&signingProxyConfig)
+
+	require.NoError(t, err)
```
Comment (fabioberger) [inline integration_test.go]: Ditto above, no need for newline between call and error check.

Comment (fabioberger) [reply integration_test.go]: Please fix everywhere.

File: integration_test.go (hunk 3 above)
Comment (opaolini) [reply integration_test.go]: Done for all

File: integration_test.go (hunk 5 above)
Comment (opaolini) [reply integration_test.go]: Done for all.

File: integration_test.go (hunk 4 above)
Comment (opaolini) [reply integration_test.go]: It did at the end of the tests, however now that is done explicitly by stopping the server with:
```
	authProxyServer.Shutdown(ctx)
	signingProxyServer.Shutdown(ctx)
```

File: integration_test.go (hunk 2 above)
Comment (opaolini) [reply integration_test.go]: It is used at the end with:
```
	es.Stop(ctx)
```

However now we also stop the proxyServers explicitly with:
```
	es.Stop(ctx)
	authProxyServer.Shutdown(ctx)
	signingProxyServer.Shutdown(ctx)
```

**Pull request merged**